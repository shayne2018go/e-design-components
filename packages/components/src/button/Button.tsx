import { computed, defineComponent } from "vue"
import { SizeProp } from "../config-provider"
import PropTypes, { tuple } from "../_util/vue-types"

const buttonVariants = tuple("base", "outline", "dashed", "text")
const buttonThemes = tuple("default", "primary", "danger", "warning", "success")

const buttonShapes = tuple("square", "round", "circle")
const ButtonHTMLTypes = tuple("submit", "button", "reset")

const buttonProps = {
  variant: PropTypes.oneOf(buttonVariants).def("base"),
  theme: PropTypes.oneOf(buttonThemes).def("default"),
  shape: PropTypes.oneOf(buttonShapes),
  size: SizeProp,
  disabled: PropTypes.bool.def(false),
  htmlType: PropTypes.oneOf(ButtonHTMLTypes).def("button"),
  loading: PropTypes.bool.def(false),
  block: PropTypes.bool.def(false),
  label: PropTypes.string,
}

export default defineComponent({
  name: "EButton",
  props: buttonProps,
  setup(props) {
    const btnClasses = computed(() => {
      return {
        [`e-btn`]: true,
        [`e-btn-${props.variant}`]: !!props.variant,
      }
    })
    return { btnClasses }
  },
  render() {
    return (
      <div class={this.btnClasses}>
        {this.$slots.default?.() || this.$props.label}
      </div>
    )
  },
})
