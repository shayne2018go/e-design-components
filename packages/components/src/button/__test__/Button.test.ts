import { shallowMount } from "@vue/test-utils"
import { expect, test } from "vitest"
import Button from "../Button"

test("mount component", () => {
  const wrapper = shallowMount(Button, {
    slots: {
      default: () => "1111111111",
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
})
