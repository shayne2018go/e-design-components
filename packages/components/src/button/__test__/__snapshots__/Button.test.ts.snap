// Vitest Snapshot v1

exports[`mount component 1`] = `"<div class=\\"button\\">1111111111</div>"`;
